src/safeposix/syscalls/fs_calls.rs

```diff
@@ -814,6 +823,48 @@ impl Cage {
         ret
     }
 
+    pub fn sbrk(&self, brk: u32) -> i32 {
+        let mut vmmap = self.vmmap.write();
+        let heap = vmmap.find_page(0).unwrap().clone();
+
+        if brk == 0 {
+            return (PAGESIZE * heap.npages) as i32;
+        }
+
+        let brk_page = ((brk + 65536 - 1) / 65536) * 16;
+
+        let heap_size = heap.npages;
+        vmmap.add_entry_with_overwrite(0, heap_size + brk_page, heap.prot, heap.maxprot, heap.flags, heap.backing, heap.file_offset, heap.file_size, heap.cage_id);
+        
+        let usr_heap_base = (heap_size * PAGESIZE) as i32;
+        let sys_heap_base = vmmap.user_to_sys(usr_heap_base)as *mut u8;
+
+        drop(vmmap);
+
+        // TODO: Currently we are not calling mmap to change prot here
```

Comment:
what does this mean?

Reply:
This is currently a TODO in wasmtime side. Since wasmtime also keeps track of the valid memory region, if we directly pass sbrk to rawposix, wasmtime will not allow to access these new memory. Current temporary approach is to both calling wasmtime memory.grow and rawposix sbrk. And since wasmtime memory.grow already did the libc::mmap, this is not required in rawposix. I'll fix this later to make memory management completely bypass wasmtime

Reply:
okay yeah. We should discuss this further since this will be somewhat of a departure. I believe we have to grow the linear memory to the full 4GB at cage launch but then map it all PROT_NONE? 
Then here for sbrk we can mprotect off chunks of it and do mmaps normally.